src/MeetingsSDKAdapter.test.js

```diff
@@ -87,33 +102,70 @@ describe('Meetings SDK Adapter', () => {
-      global.console.error = jest.fn();
-      const localAudio = await meetingSDKAdapter.getStream(meetingID, {sendAudio: true});
+    test('throws errors, nullifies local Audio and sets permissions to denied, dismissed or error if not retrieved successfully', (done) => {
+      const permissions = [
+        'DENIED',
+        'DISMISSED',
+        'ERROR',
+      ];
+      const constraint = {sendAudio: true};
 
-      expect(localAudio).toBeNull();
-      expect(global.console.error).toHaveBeenCalledWith(
-        expect.stringContaining('Unable to retrieve local media stream for meeting "meetingID"'),
-        undefined,
+      global.console.error = jest.fn();
+      mockSDKMeeting.getMediaStreams = jest.fn(() => Promise.reject());
+      meetingSDKAdapter.getStream(meetingID, constraint).pipe(last()).subscribe(
+        ({permission, stream}) => {
+          expect(stream).toBeNull();
+          expect(permissions.indexOf(permission)).toBeGreaterThan(-1);
+          expect(global.console.error).toHaveBeenCalledWith(
+            `Unable to retrieve local media stream for meeting "meetingID" with constraint "${constraint}"`, undefined,
+          );
+          done();
+        },
       );
     });
 
-    test('throws errors and nullifies local Video if not retrieved successfully', async () => {
-      mockSDKMeeting.getMediaStreams = jest.fn(() => Promise.reject());
-      global.console.error = jest.fn();
-      const localVideo = await meetingSDKAdapter.getStream(meetingID, {sendVideo: true});
+    test('throws errors, nullifies local Video and sets permissions to denied, dismissed or error if not retrieved successfully', (done) => {
+      const permissions = [
+        'DENIED',
+        'DISMISSED',
+        'ERROR',
+      ];
 
-      expect(localVideo).toBeNull();
-      expect(global.console.error).toHaveBeenCalledWith(
-        expect.stringContaining('Unable to retrieve local media stream for meeting "meetingID"'),
-        undefined,
+      mockSDKMeeting.getMediaStreams = jest.fn(() => Promise.reject());
+      meetingSDKAdapter.getStream(meetingID, {sendVideo: true}).pipe(
+        last(),
+      ).subscribe(
+        ({permission, stream}) => {
+          expect(stream).toBeNull();
+          expect(permissions.indexOf(permission)).toBeGreaterThan(-1);
+          done();
+        },
+        (error) => {
+          expect(error.message).toBe('Unable to retrieve local media stream for meeting "meetingID"');
+          done();
+        },
       );
     });
 
-    test('returns local media in a proper shape', async () => {
-      expect(await meetingSDKAdapter.getStream(meetingID, {sendAudio: true})).toEqual(['localAudio']);
-      expect(await meetingSDKAdapter.getStream(meetingID, {sendVideo: true})).toEqual(['localVideo']);
+    test('returns local media in a proper shape', (done) => {
+      meetingSDKAdapter.getStream(meetingID, {sendAudio: true}).pipe(
```

Comment:
consider this indentation:
```js
meetingSDKAdapter.getStream(meetingID, {sendAudio: true})
  .pipe(last())
  .subscribe((localMedia) => {
    expect(...);
    done();
  });
```

if you find it more readable, please use it, otherwise keep the existing one